### ISSUE TITLE
mutationType not working on type policies declaration

### ISSUE THREAD

**Issue by zaguiini** [apollographql/apollo-client#7443]
**Intended outcome:**
To use a different mutation root type than `Mutation`.

**Actual outcome:**
Setting `mutationType` to `true` on the custom mutation type does not make any difference and Apollo attempts to write cache to `Mutation` instead of the desired one.

**How to reproduce the issue:**
It's proprietary code. I'll try my best to describe.
Root mutation is called `TalentMutation`. I've set:

```ts
{
  TalentMutation: {
    mutationType: true,
    fields: {
      talentPortalSetting: { merge: false }
    }
  }
}
```

Throws the following warning:

```To address this problem (which is not a bug in Apollo Client), either ensure all objects of type TalentPortalSettingOps have an ID or a custom merge function, or define a custom merge function for the Mutation.talentPortalSetting field, so InMemoryCache can safely merge these objects.```

**Versions**
```
  System:
    OS: macOS 10.15.7
  Binaries:
    Node: 14.8.0 - ~/.nvm/versions/node/v14.8.0/bin/node
    Yarn: 1.22.5 - ~/.nvm/versions/node/v14.8.0/bin/yarn
    npm: 5.1.0 - ~/Development/toptal/talent-portal-frontend/node_modules/.bin/npm
  Browsers:
    Chrome: 87.0.4280.88
    Edge: 87.0.664.57
    Firefox: 83.0
    Safari: 14.0
  npmPackages:
    @apollo/client: ^3.3.4 => 3.3.4
    apollo-upload-client: ^14.1.3 => 14.1.3
```

**Additional information**

It's properly working on `3.2.9`. Not sure what's changed. I haven't seen anything relevant on the CHANGELOG, so it must be something changed by accident.

**Comment by zaguiini** [comment]
Hey @benjamn, it was `3.3.0-beta.3` which introduced the regression. I'll try and find what went wrong.

**Comment by benjamn** [comment]
@zaguiini Thanks! That's a very useful clue for my investigation. 🙏

**Comment by benjamn** [comment]
This should be fixed in `@apollo/client@3.3.6` (just published). Please feel free to reopen if you're still having issues. Thanks!
